spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/watcher/DependencyWatcher.java

```diff
@@ -0,0 +1,34 @@
+package org.springframework.cloud.zookeeper.discovery.watcher;
+
+import java.io.IOException;
+
+public interface DependencyWatcher {
```

Comment:
In fact they can only register listeners and not hooks.

We could split it into two interfaces and that would make more sense.